### ISSUE TITLE
Integration Tests OS X: Incomplete Authorizations Error

### ISSUE THREAD

**Issue by BKreisel** [letsencrypt/letsencrypt#857]
Integrations failing on native OS X on what I believe to be the challenge response section. 

Log Snapshot:
```
2015-09-28 20:24:32,825:WARNING:acme.challenges:Using non-standard port for SimpleHTTP verification: 5001
2015-09-28 20:24:32,825:DEBUG:acme.challenges:Verifying simpleHttp at https://le.wtf:5001/.well-known/acme-challenge/bNcpUsnOTIRG63kH0lZd6al6IgZeTaXXBfSn_xcOei4...
2015-09-28 20:24:32,828:INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): le.wtf
/Users/brandon/Projects/letsencrypt/venv/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:768: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.org/en/latest/security.html
  InsecureRequestWarning)
127.0.0.1 - - [28/Sep/2015 20:24:33] "GET /.well-known/acme-challenge/bNcpUsnOTIRG63kH0lZd6al6IgZeTaXXBfSn_xcOei4 HTTP/1.1" 200 -
2015-09-28 20:24:33,069:DEBUG:requests.packages.urllib3.connectionpool:"GET /.well-known/acme-challenge/bNcpUsnOTIRG63kH0lZd6al6IgZeTaXXBfSn_xcOei4 HTTP/1.1" 200 915
2015-09-28 20:24:33,081:DEBUG:acme.challenges:Received <Response [200]>: -n {....<omitted>..}
2015-09-28 20:24:33,081:DEBUG:acme.challenges:Deserialization error: No JSON object could be decoded
2015-09-28 20:24:33,082:ERROR:letsencrypt.plugins.manual:Self-verify of challenge failed, authorization abandoned.
2015-09-28 20:24:33,082:INFO:letsencrypt.auth_handler:Waiting for verification...
2015-09-28 20:24:33,082:INFO:letsencrypt.auth_handler:Cleaning up challenges
2015-09-28 20:24:33,082:DEBUG:letsencrypt.plugins.manual:Terminating manual command process
2015-09-28 20:24:33,083:DEBUG:letsencrypt.cli:Exiting abnormally:
```

Stacktrace:
```
Traceback (most recent call last):
  File "/Users/brandon/Projects/letsencrypt/venv/bin/letsencrypt", line 9, in <module>
    load_entry_point('letsencrypt==0.1', 'console_scripts', 'letsencrypt')()
  File "/Users/brandon/Projects/letsencrypt/letsencrypt/cli.py", line 962, in main
    return args.func(args, config, plugins)
  File "/Users/brandon/Projects/letsencrypt/letsencrypt/cli.py", line 370, in auth
    _auth_from_domains(le_client, config, domains, plugins)
  File "/Users/brandon/Projects/letsencrypt/letsencrypt/cli.py", line 291, in _auth_from_domains
    lineage = le_client.obtain_and_enroll_certificate(domains, plugins)
  File "/Users/brandon/Projects/letsencrypt/letsencrypt/client.py", line 231, in obtain_and_enroll_certificate
    certr, chain, key, _ = self.obtain_certificate(domains)
  File "/Users/brandon/Projects/letsencrypt/letsencrypt/client.py", line 214, in obtain_certificate
    return self._obtain_certificate(domains, csr) + (key, csr)
  File "/Users/brandon/Projects/letsencrypt/letsencrypt/client.py", line 172, in _obtain_certificate
    authzr = self.auth_handler.get_authorizations(domains)
  File "/Users/brandon/Projects/letsencrypt/letsencrypt/auth_handler.py", line 87, in get_authorizations
    self.verify_authzr_complete()
  File "/Users/brandon/Projects/letsencrypt/letsencrypt/auth_handler.py", line 300, in verify_authzr_complete
    raise errors.AuthorizationError("Incomplete authorizations")
AuthorizationError: Incomplete authorizations
```

**Comment by kuba** [comment]
The following line is concerning:
```
2015-09-28 20:24:33,081:DEBUG:acme.challenges:Received <Response [200]>: -n {....<omitted>..}
2015-09-28 20:24:33,081:DEBUG:acme.challenges:Deserialization error: No JSON object could be decoded
```
What is this `-n` doing there? Hm...

In general, seems like the automatically executed manual plugin command fails to set up HTTP server that correctly provisions needed resource. Could you please run `letsencrypt -a manual -d le.wtf auth`, copy paste the command to another terminal, and try to access `https://le.wtf:5001/.well-known/acme-challenge/$token` in your browser / `curl`? Are you sure `le.wtf` resolves to `127.0.0.1`, did you set up an appropriate `/etc/hosts` entry?

**Comment by BKreisel** [comment]
So the -n was due to the fact that `subprocess.Popen` uses `sh` as the default execution shell. OS X's `sh` doesn't ship with the `-n` flag for `echo`. This muddled up the JSON deserialization. After switching the executable to `/bin/bash` all seems to be well

**Comment by BKreisel** [comment]
its now failing later on in the integration tests with the following stacktrace:

```python
2015-10-03 11:43:10,296:DEBUG:acme.client:Serialized JSON: {"resource": "new-cert", "csr": "MIICajCCAVICAQAwADCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBALUyjT1X9Ggc467VObAf5SvOkNvaZ8Q86Ca1Qh2WzdmR1Tm0dFTiz0baOGOfHNJfuwdsg0nBfAjVyk0EZLJoWLamGviGf7MFGytnaYT3mlQyqd2knCt4MmFT3LlPwTMzk0WHsJ-CkfFolSyCZkXXGnKPXiYoKfYhFzuVmXNpLFJTvOd_lOJMWjW80aNmqxao72YVPnpEy0SeJsVpvz0pEkWY53O5xqlrvHpzXAUPgb0F6jMCb8r6jq7SEJ4cQQkhQ9lOx74S7oB-x1_nO_pjqrddE6XfnQvga-Hbe5PMq81TVXTSBgpo887CC-P-_fQqFFro05g1osnOBqLTxnAZSMkCAwEAAaAlMCMGCSqGSIb3DQEJDjEWMBQwEgYDVR0RBAswCYIHbGUzLnd0ZjANBgkqhkiG9w0BAQUFAAOCAQEAksfUciQqJcnX6-SXSfdKiimJd_yqsdAr6QxDT--mb8FVGdbJRaEdPoVGv0z7l0PH68R40tt2MilQCqmeqbZoLx4CcP90fpG1Y30orS7Qo5JPs1v5IthOqWSRM01l4WiI_TlctZ5RAWJ2N3m6V2AY6xutI8DnIwBiHwZi7EXVVhq4CbNTXaKaS37QoHsru7aW3mUMBvr5nGIOof-1k4gzq4hyr8JIVz9FWjPCUVqEGxIMbNUfY0mOVOHn2pf3iBA6j4GjiR2PFXsDfJ2OQK-k-dzcDziYmOb03WRQQcIks97BCDht2ONYPz8M81g0Vk178Wx8IG73f8I5_edqwI4HPA"}
2015-10-03 11:43:10,297:DEBUG:acme.jose.json_util:Omitted empty fields: typ=None, x5c=(), crit=(), x5tS256=None, kid=None, cty=None, jwk=None, x5u=None, x5t=None, alg=None, jku=None
2015-10-03 11:43:10,296:DEBUG:letsencrypt.error_handler:Encountered exception:
Traceback (most recent call last):
  File "/Users/brandon/Projects/letsencrypt/letsencrypt/auth_handler.py", line 115, in _solve_challenges
    dv_resp = self.dv_auth.perform(self.dv_c)
  File "/Users/brandon/Projects/letsencrypt/letsencrypt/plugins/standalone/authenticator.py", line 346, in perform
    if self.start_listener(self.config.dvsni_port):
  File "/Users/brandon/Projects/letsencrypt/letsencrypt/plugins/standalone/authenticator.py", line 289, in start_listener
    return self.do_child_process(port)
  File "/Users/brandon/Projects/letsencrypt/letsencrypt/plugins/standalone/authenticator.py", line 227, in do_child_process
    self.connection, _ = self.sock.accept()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/socket.py", line 206, in accept
    sock, addr = self._sock.accept()
  File "/Users/brandon/Projects/letsencrypt/letsencrypt/plugins/standalone/authenticator.py", line 116, in subproc_signal_handler
    sys.exit(0)
SystemExit: 0
```
Still trying to root out the issue here

**Comment by kuba** [comment]
Please update to `master` before further debugging. Line 116 is not the same as in your output. Stuff might have been fixed with #868.

**Comment by bmw** [comment]
Fixed in #886.
